### PR TITLE
PLAT-72737: TimePickerBase - Update lifecycle method

### PR DESCRIPTION
**Pull request by teckliew**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Update lifecycle in `TimePickerBase`

### Links
[//]: # (Related issues, references)
PLAT-72737

### CODE REVIEW

**Comment by teckliew** [comment]
Sounds good, I simplified `HourPicker` with `shouldComponentUpdate`. LMK WYT

**Comment by viodragon2** [comment]
Shoot. I now understand why it was setting `noAnimation` momentarily. By blocking the render, it's not going to update the position of the value. You can see the problem by updating the hour after changing meridiem picker. (e.g. 5AM -> 5PM -> press up in hour picker -> 6AM)

File: packages/moonstone/TimePicker/TimePickerBase.js [hunk 1]
```diff
@@ -43,18 +43,21 @@ class HourPicker extends React.Component {
 		};
```
Comment (viodragon2) [inline packages/moonstone/TimePicker/TimePickerBase.js]: Need to add a default value for `prevValue` state

File: packages/moonstone/TimePicker/TimePickerBase.js [hunk 2]
```diff
@@ -43,18 +43,21 @@ class HourPicker extends React.Component {
...
-	UNSAFE_componentWillReceiveProps (nextProps) {
-		const {children, value} = this.props;
-		const {children: nextChildren, value: nextValue} = nextProps;
+	static getDerivedStateFromProps (props, state) {
+		const hours = props.hasMeridiem ? hours12 : hours24;
 
-		this.setState({
-			noAnimation: children[value] === nextChildren[nextValue]
-		});
+		return {
```
Comment (viodragon2) [inline packages/moonstone/TimePicker/TimePickerBase.js]: Let's add a condition to return `null` when there's no change in the value.

e.g.
```

		if (state.prevValue !== props.value) {
			return {
				noAnimation: hours[state.prevValue] === hours[props.value],
				prevValue: props.value
			};
		}
		return null;
```

**Pull request merged**